test/sanity/validate-modules/main.py

```diff
@@ -1011,7 +1013,17 @@ def _validate_argument_spec(self):
             )
             return
 
+        provider_args = set()
+        args_from_argspec = set()
         for arg, data in spec.items():
+            args_from_argspec.add(arg)
+            args_from_argspec.update(data.get('aliases', []))
+            if arg == 'provider':
```

Comment:
Shouldn't this be constrained to network modules, to avoid a false hit on the arg in non-network modules?

Reply:
Do you have a specific recommendation on detecting that?

Reply:
Checking to see if the module resides under the `lib/ansible/modules/network/` directory should work.

Reply:
This has been addressed.